xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingChecksum.java

```diff
@@ -0,0 +1,35 @@
+package info.bitrich.xchangestream.kraken;
+
+import org.knowm.xchange.dto.trade.LimitOrder;
+
+import java.math.BigDecimal;
+import java.util.TreeSet;
+import java.util.zip.CRC32;
+
+public class KrakenStreamingChecksum {
+    static void addBigDecimalToCrcString(StringBuilder stringBuilder, BigDecimal bigDecimal) {
+        stringBuilder.append(bigDecimal.movePointRight(bigDecimal.scale()));
+    }
+
+    static void addOrderToCrcString(StringBuilder stringBuilder, LimitOrder order) {
+        addBigDecimalToCrcString(stringBuilder, order.getLimitPrice());
+        addBigDecimalToCrcString(stringBuilder, order.getOriginalAmount());
+    }
+
+    public static String createCrcString(TreeSet<LimitOrder> asks, TreeSet<LimitOrder> bids) {
+        StringBuilder stringBuilder = new StringBuilder();
+        asks.stream().limit(10).forEach(o -> addOrderToCrcString(stringBuilder, o));
+        bids.stream().limit(10).forEach(o -> addOrderToCrcString(stringBuilder, o));
```

Comment:
is the checksum only on top 10

Reply:
Correct, as described on https://docs.kraken.com/websockets/#book-checksum .

Reply:
I suggest to make it more obvious and extract `10` into constant with descriptive name